theforeman.org/pipelines/test/foreman/unit.groovy

```diff
@@ -0,0 +1,62 @@
+def ruby_versions = foreman_unit_versions[ghprbTargetBranch]['ruby']
+
+pipeline {
+    agent none
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('matrix') {
+            matrix {
+                agent { label 'fast' }
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.7', '2.5'
```

Comment:
I wonder if you can use the `ruby_versions` variable instead.
```suggestion
                        values ruby_versions
```

But I vaguely recall you can't and need to resort to something like https://www.jenkins.io/blog/2019/12/02/matrix-building-with-scripted-pipeline/

Reply:
Correct, what you pass to this cannot be dynamic, it has to be static.